test/deprecated.jl

```diff
@@ -0,0 +1,174 @@
+@testset "deprecated" begin
+    a = allowmissing(1.0:1.0:20.0)
+    a[[2, 3, 7]] .= missing
+    mask = map(!ismissing, a)
+
+    @testset "Drop" begin
+        result = impute(a, :drop; limit=0.2)
+        expected = copy(a)
+        deleteat!(expected, [2, 3, 7])
+
+        @test result == expected
+
+        # Mutating method
+        a2 = copy(a)
+        Impute.drop!(a2; limit=0.2)
+        @test a2 == expected
+    end
+
+    @testset "Interpolate" begin
+        result = impute(a, :interp; limit=0.2)
+        @test result == collect(1.0:1.0:20)
+        @test result == interp(a)
```

Comment:
I wish this were `interpolate` rather than `interp`.

It's more user-friendly to use whole words (especially for non-native speakers), and fits with the patten of the function being the lower-case of the type. 

Since this next version introduces deprecations, now might be a good time to make the change

Reply:
`interp` is pretty common in other languages (e.g., python, matlab, R), but I guess we have inadvertently created a convention between the types and functions.

Reply:
#28